pkg/pillar/cas/cas.go

```diff
@@ -0,0 +1,66 @@
+package cas
+
+import (
+	"fmt"
+	"io"
+	"time"
+)
+
+//CAS: provides methods to interact with CAS clients
+type CAS interface {
+	//Blob APIs
+	//CheckBlobExists: returns true if the blob exists.
+	CheckBlobExists(blobSha256 string) bool
+	//GetBlobSize: returns size of the blob
+	GetBlobSize(blobSha256 string) (int64, error)
+	//ListBlobs: returns list of blob sha256
+	ListBlobs() ([]string, error)
+	//IngestBlob: ingests a blob from the given reader and assigns the given sha256 to it.
+	IngestBlob(blobSha256 string, reader io.Reader) error
+	//ReadBlob: returns a reader of blob
+	ReadBlob(sha256 string) (io.Reader, error)
```

Comment:
* Change the args to be consistent with the other funcs that accept a hash: `<algo>:<hash>`, name the same as however we set the others
* In comment, make it something more readable: 

```go
// ReadBlob returns a reader to consume the raw data
```

* In the comment, make it clear that if the blob does not exist, it will return an error. We probably should create a specific error type for it, so they can check

Reply:
Done.